wp-content/themes/debtcollective/src/js/global/admin-bar.js

```diff
@@ -0,0 +1,20 @@
+const addTopMargin = function () {
+	const wpadminbar = document.getElementById('wpadminbar');
+	if (wpadminbar) {
+		const dcheader = document.getElementsByTagName('dc-header');
+		const styleTag = `
+			<style>
+				:host(dc-header) .navbar-top {margin-top:46px;}
+				@media (min-width: 782px) {
+					:host(dc-header) .navbar-top {margin-top:32px;}
+				}
+			</style>
+		`;
+		const combinedHtml = styleTag.concat(dcheader[0].shadowRoot.innerHTML);
+		dcheader[0].shadowRoot.innerHTML = combinedHtml;
+	}
+};
+
+document.addEventListener('DOMContentLoaded', function () {
```

Comment:
@robu9 - good stuff all around.

We should also set an onresize listener, and call it on domcontentloaded.
So that we 

```
document.addEventListener('DOMContentLoaded', function () {
    window.doResizeStuff = function() {
       // do resize stuff
    }

    // dostuff
    …
    
    window.onresize = window.doResizeStuff
    // fire resize event
    window.dispatchEvent(new Event('resize'));
  };
```

Reply:
@thiagodemellobueno Hey thinking about this - I do have a media query in the style tag to set different top margins based on what size the wp admin bar is (since it's taller on smaller screens) - is that sufficient? If not I think I'm missing something/not understanding - unsure what should be happening in the resize function if you could help me out there/elaborate a little more. Thanks!

Reply:
Well, if you are setting sizes based on the height of an element, on pageload -  and that window changes size it becomes necessary to recalculate any variables that were dependent on certain window dimentions. Or you are working with values that were true on page load but may have changed. It's always good if you are using onresize stuff, to run that once the dom is loaded (DocumentReady). 

If you are setting the sizes in cssexclusively using media tags, then why are we doing stuff in JS at all?
If its unecessary, lets remove it.

If you are setting that 32px hard coded instead of using the toolbar size (https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetHeight), then we should change that. Also if you are using media queries, then which breakpoint is the 32px you are setting in this code for, and isnt the css just handling it anyway?

Reply:
Definitely - that makes sense. The reason we're using js to add the css is because everything in the `dc-header` component is shadow DOM so the style rules we write in the usual way don't get applied. So what I'm doing is creating a style tag with the desired rules (including the media query) in js, and adding it to the html in the shadow tree. Admittedly, I'm not super well versed on shadow DOM stuff, so if there's a better way to do this I'm open to learning about it and doing this however would be best - this is the first solution I got to work so I went with it. On smaller screens a margin-top of 46px is needed to offset the height of the wp admin bar, and on screens larger than 782px the wp admin bar becomes 32px tall and that's how much top margin is needed. As it currently is, if the user resizes the page after page load the media query handles the different margin rules like you'd expect it to.